### PR TITLE
BRAKING MODEL CHANGES

### PR DESCRIPTION
**Pull request by ntrncic**
Refactor intervention api

### CODE REVIEW

**Comment by ilcsik** [comment]
I was working through the fields and a couple of questions came up:

1, Do we save reference number as a field or leave it as calculated on the fly?
2, There are cancelled and in process statuses that are not in the requirements doc (and draft and suspended which is not in the model). Do we need the 2 that are not in the req? 
3, What endpoints do we need exactly? I see a standalone /interventions endpoint, do we need detail as well, or is it mostly used through partner/pk/interventions etc?

**Comment by ilcsik** [comment]
Some more questions:

4, How 'Locations' field  should work? M2M?
5, Whats 'Humanitarian response plan' field?
6, Planned visits section: which field is writable? Whats 'partner risk rating' and 'audit'?
7, 'Fr number' is charfield currently, is that ok?

**Comment by ntrncic** [comment]
@ilcsik 
1. There is already a field on the model called `number` that it's save to
2. Will add the statuses. The other ones should remain there until we can do a data migration
3. We need interventions api by itself. therefore needs detail view. It will not be updates through partners/interventions
4. There is an intermediary model `GwPCALocation`
5. This field is actually `result_structure` on the PCA model
6. Planned Visits will be it's own model with foreign key to PCA
7.  Added `fr_numbers` array field to the PCA model

**Comment by robertavram** [comment]
@ntrncic @ilcsik drop the work on this api for now.. interventions model needs to be cleaned and adjusted. I will explain everything in comments.

@ilcsik see if you can take the latest issue assigned to you

**Comment by ilcsik** [comment]
Comments and answers (except 6. Planned visits) addressed in https://github.com/unicef/etools/pull/280/commits/239b583cbc05d9c718f6ad3677dfa29caad769a0

**Comment by ilcsik** [comment]
I think this PR now reflects what's in the requirements in terms of views/fields/validation etc. except programme results (which is being refactored?).

It needs some in depth review though since it's mostly a huge chunk of nested models with similar serialization and CRUD methods. I suppose there's room for some fine tuning for e.g in terms of data representation, or disallowing part of CRUD on some models etc.

**Comment by ntrncic** [comment]
@ilcsik please redo the views/fields/validation and including the Results model for outputs. We have changed the PCA model to a new model named Intervention. The api/v2 will look at this model only.
Please take a look and make changes to the views and serializers and validations as needed. There are new fields and some field names have changed on the new model

**Comment by ilcsik** [comment]
@ntrncic ok I'm on it

**Comment by ilcsik** [comment]
@robertavram In absence of Nik I'm asking you: I'm trying to get this branch work after model changes to do the api refactor but I have an error, I'm not sure if this field should exists or shouldnt (error is during running tests):

ProgrammingError: column partners_agreementamendmentlog.signed_document does not exist

Can you please advise?

**Comment by robertavram** [comment]
@ilcsik dude.. always feel free to ask me even if Nik is there. in order to make sure I get to it quickly though.. please personal slack me a link to your comment...
This branch is a huge mess... has a whole bunch of problems...
Join me tomorrow on a call if you can for a couple of hours and we'll work on it together.... drop this branch until then...  I just assigned you an issue you could take a look at until then

File: EquiTrack/partners/forms.py [hunk 1]
```diff
@@ -297,6 +271,14 @@ class AgreementForm(UserGroupForm):
     user_field = u'signed_by'
     group_name = u'Senior Management Team'
 
+    def __init__(self, *args, **kwargs):
+        super(AgreementForm, self).__init__(*args, **kwargs)
+        if kwargs.get('instance', None) and kwargs["instance"].partner:
+            self.fields['authorized_officers'].queryset = PartnerStaffMember.objects.filter(partner=kwargs["instance"].partner)
+        else:
+            self.fields['authorized_officers'].disabled = True
+            self.fields['authorized_officers'].queryset = PartnerStaffMember.objects.none()
```
Comment (robertavram) [inline EquiTrack/partners/forms.py]: fantastic!

**Pull request merged**